Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/DataEditorForm.java

```diff
@@ -276,16 +280,90 @@ public String save() {
                 ServiceManager.getMetsService().save(workpiece, out);
             }
             return close();
-        } catch (InvalidMetadataValueException | IOException | NoSuchMetadataFieldException e) {
+        } catch (InvalidMetadataValueException | IOException | NoSuchMetadataFieldException | DAOException
+                | DataException e) {
             Helper.setErrorMessage(e.getLocalizedMessage(), logger, e);
             return null;
         }
     }
 
+    /**
+     * Updates the child links in the database.
+     *
+     * @throws DAOException
+     *             if a HibernateException is thrown
+     * @throws DataException
+     *             if an IOException is thrown
+     */
+    private void updateChildLinks() throws DAOException, DataException {
+        Set<Integer> referencedChildren = new TreeSet<>();
+        addAllChildrenRecursive(workpiece.getRootElement(), referencedChildren);
+        TreeSet<Integer> savedChildren = new TreeSet<>();
+        process.getChildren().parallelStream().map(Process::getId).forEachOrdered(savedChildren::add);
+        Iterator<Integer> referencedChildrenIterator = referencedChildren.iterator();
+        Iterator<Integer> savedChildrenIterator = savedChildren.iterator();
+        boolean nextReferencedChild = true;
+        boolean nextSavedChild = true;
+        Integer referencedChildId = null;
+        Integer savedChildId = null;
+        boolean save = false;
+        do {
+            if (nextReferencedChild) {
+                referencedChildId = referencedChildrenIterator.hasNext() ? referencedChildrenIterator.next() : null;
+                nextReferencedChild = false;
+            }
+            if (nextSavedChild) {
+                savedChildId = savedChildrenIterator.hasNext() ? savedChildrenIterator.next() : null;
+                nextSavedChild = false;
+            }
+            if (Objects.nonNull(referencedChildId) && Objects.isNull(savedChildId)
+                    || referencedChildId < savedChildId) {
+                Process childToAdd = ServiceManager.getProcessService().getById(referencedChildId);
+                childToAdd.setParent(process);
+                process.getChildren().add(childToAdd);
+                save = true;
+                nextReferencedChild = true;
+            } else if (Objects.isNull(referencedChildId) && Objects.nonNull(savedChildId)
+                    || referencedChildId > savedChildId) {
+                Process childToRemove = ServiceManager.getProcessService().getById(savedChildId);
+                childToRemove.setParent(null);
+                process.getChildren().remove(childToRemove);
+                save = true;
+                nextSavedChild = true;
+            } else if (Objects.nonNull(referencedChildId) && Objects.nonNull(savedChildId)) {
+                nextReferencedChild = true;
+                nextSavedChild = true;
+            }
+        } while (nextReferencedChild || nextSavedChild);
```

Comment:
Do you need those flags, or is a check for NonNull on th id's sufficient? Because as I see it, this dependent